src/main/java/seedu/address/model/entity/Team.java

```diff
@@ -127,6 +127,107 @@ public void setLocation(Location location) {
 
      */
 
+    /**
+     * Updates the participant in the team.
+     *
+     * @param updatedParticipant
+     * @return boolean
+     */
+    public boolean updateParticipant(Participant updatedParticipant) {
+        List<Participant> list = this.getParticipants();
+        boolean isParticipantInTeam = false;
+        for (int i = 0; i < list.size(); i++) {
+            if (list.get(i).getId().equals(updatedParticipant.getId())) {
+                list.set(i, updatedParticipant);
+                isParticipantInTeam = true;
+            }
+        }
+        return isParticipantInTeam;
+    }
+
+    /**
+     * Adds a participant into the team.
+     *
+     * @param participant
+     * @return boolean
+     */
+    public boolean addParticipant(Participant participant) {
+        List<Participant> list = this.getParticipants();
+        if (list.contains(participant)) {
+            return false;
+        }
+        list.add(participant);
+        this.setParticipants(list);
+        return true;
+    }
+
+    /**
+     * Deletes the participant from the team if it exists.
+     *
+     * @param participant
+     * @return boolean
+     */
+    public boolean deleteParticipant(Participant participant) {
+        if (!this.participants.contains(participant)) {
+            return false;
+        }
+        this.participants.remove(participant);
+        return true;
+    }
+
+    /**
+     * Updates the mentor in the team if it exists.
+     *
+     * @param updatedMentor
+     * @return boolean
+     */
+    public boolean updateMentor(Mentor updatedMentor) {
+        if (this.mentor.isEmpty()) {
+            return false;
+        }
+
+        if (!this.mentor.get().getId().equals(updatedMentor.getId())) {
+            return false;
+        }
+
+        this.mentor = Optional.of(updatedMentor);
+        return true;
+    }
+
+    /**
+     * Adds a mentor to the team.
+     *
+     * @param mentor
+     * @return
+     */
+    public boolean addMentor(Mentor mentor) {
+        Optional<Mentor> mentorOptional = this.getMentor();
+        if (!mentorOptional.isEmpty()) {
+            return false;
+        }
+        this.setMentor(Optional.of(mentor));
+        return true;
+    }
+
+    /**
+     * Deletes the mentor if it exists.
+     *
+     * @param mentor
+     * @return
+     */
+    public boolean deleteMentor(Mentor mentor) {
```

Comment:
Why would we need to pass the mentor as a parameter? And I think we should be consistent with the return types and return the Mentor deleted

Reply:
Ah no this is called ineternally by modelmanager, the modelmanager one still returns mentor

Reply:
This method is used for the cross validation bit

Reply:
oh cool cool